### DISCUSSION TITLE
I set my own umap (with random_state=0), hdbscan (with random_state=0), ctfidf_model, and representation_model, and then passed these four modules into the BERTopic() function. However, my results are different every time I run the code. What should I do?

### DISCUSSION THREAD

**Discussion by superseanyoung** [MaartenGr/BERTopic#2214] [General]
topic_model = BERTopic(embedding_model=transformer_model, 
                           min_topic_size=3, 
                           verbose=True,
                           umap_model=umap_model,
                           hdbscan_model=hdbscan_model,
                           ctfidf_model=ctfidf_model,
                           representation_model=representation_model,
                           #top_n_words=10,
                           #min_topic_size=10,
                           #nr_topics=None,
                           #low_memory=False,
                           #calculate_probabilities=True
                          )
    topics,probs= topic_model.fit_transform(sentences,embeddings=embeddings)

**Comment by MaartenGr** [comment]
Could you create an issue for this as a bug and fill in all the information listed there? It's difficult to say what is happening without more context. Also, make sure you provide the **full** code.

**Reply by superseanyoung** [reply]
As long as the "random_state" parameter in the UMAP model is set, can it make the entire model non-random?

**Reply by MaartenGr** [reply]
Perhaps, but as mentioned I will need more context. I have no clue what's inside any of the models you passed to BERTopic, the version you are using, the OS between multiple runs, etc. Could you follow the steps I mentioned above?

**Reply by superseanyoung** [reply]
I solved the problem and found that as soon as you set random_state in umap, the randomness of the model is controlled, and the randomness of the subsequent outliers is controlled.

**Reply by superseanyoung** [reply]
Since I don't think this is a bug or anything, I just wanted to discuss it with you,

**Reply by MaartenGr** [reply]
That is correct, the `random_state` in UMAP it *typically* the one that controls the randomness but that might change if you use different submodels.
